### PR TITLE
Update composite layers to use multiple prop objects

### PR DESCRIPTION
**Pull request by ibgreen**
Hex and Grid layer are using more complicated logic, will defer changing these.

### CODE REVIEW

File: src/core/lifecycle/create-props.js [hunk 1]
```diff
@@ -62,6 +62,12 @@ function getDefaultProps(layerClass) {
   }
 
   const parent = layerClass.prototype;
+  if (!parent) {
+    return {
```
Comment (Pessimistress) [inline src/core/lifecycle/create-props.js]: `return null` should work

File: src/core-layers/geojson-layer/geojson-layer.js [hunk 2]
```diff
@@ -144,7 +144,9 @@ export default class GeoJsonLayer extends CompositeLayer {
       hasPolygon &&
       new subLayers.PolygonLayer(
         this.getSubLayerProps({
-          id: 'polygon-fill',
+          id: 'polygon-fill'
+        }),
```
Comment (Pessimistress) [inline src/core-layers/geojson-layer/geojson-layer.js]: `CompositeLayer. getSubLayerProps` also merges `updateTriggers`.

File: src/core/lifecycle/create-props.js (hunk 1 above)
Comment (ibgreen) [reply src/core/lifecycle/create-props.js]: > return null should work

I think it would break on line 11.

Comment (Pessimistress) [reply src/core/lifecycle/create-props.js]: Assuming `createProps` is always called by the `Layer` class, line 11 will not get `null`.

Comment (ibgreen) [reply src/core/lifecycle/create-props.js]: You are probably right. I'd prefer to let it stand as is, just to minimize churn. The reason is that the code is safe/working as it is, and I just confirmed that this line will be changed in successive, staged diffs anyway. 

My assumption is that the main reason you requested changes was the issue with the `updateTriggers` forwarding, which I addressed, not this point, right?

**Pull request merged**